internal/nginx/config/generator.go

```diff
@@ -150,6 +174,60 @@ func generateProxyPass(address string) string {
 	return "http://" + address
 }
 
+var supportedFilters = map[v1beta1.HTTPRouteFilterType]struct{}{
+	v1beta1.HTTPRouteFilterRequestRedirect: {},
+}
+
+func findFirstFilters(
+	filters []v1beta1.HTTPRouteFilter,
+	filterTypes map[v1beta1.HTTPRouteFilterType]struct{},
+) map[v1beta1.HTTPRouteFilterType]v1beta1.HTTPRouteFilter {
+
+	result := make(map[v1beta1.HTTPRouteFilterType]v1beta1.HTTPRouteFilter)
+	for i := len(filters) - 1; i >= 0; i-- {
```

Comment:
Would we gain much performance-wise by iterating from `0-n` and returning the first occurrence of a filter?

I can see that this approach will work for multiple filter types but it means that we always iterate over the full list of filters even when it's not necessary.

Reply:
I simplified it since we only support one filter type for now